src/SASViyaApiClient.ts

```diff
@@ -381,7 +387,17 @@ export class SASViyaApiClient {
         )
         const newFolderName = `${parentFolderPath.split('/').pop()}`
         if (newParentFolderPath === '') {
```

Comment:
It would be great if we could extract this entire section into a function and add test(s) for it.